src/electron/routing_service.ts

```diff
@@ -323,7 +323,7 @@ async function installLinuxRoutingServices(): Promise<void> {
   command +=
     ' && ' +
     installationFileDescriptors
-      .map(({filename, sha256}) => `/usr/bin/echo "${sha256}  ${path.join(tmp, filename)}" | /usr/bin/shasum -a 256 -c`)
+      .map(({filename, sha256}) => `/usr/bin/echo "${sha256}  ${path.join(tmp, filename)}" | (/usr/bin/shasum -a 256 -c || /usr/bin/core_perl/shasum -a 256 -c)`)
```

Comment:
This is not really the logic we want. If the first command fails for whatever reason (e.g. fingerpring fails) we don't want to run the second command. And we don't want to try to run the first command if it doesn't exist. It's a lot cleaner and easier to reason about if you check the existence of the files before running them.

This also needs to be tested to make sure it works on Ubuntu/Debian, which is our only supported Linux.

/cc @jyyi1 to review

Reply:
Agreed, I'd rather test which `shasum` (either `/usr/bin/shasum` or `/usr/bin/core_perl/shasum`) exists before running the command, and only append the correct full `shasum` path to the `command`:

```typescript
shasumCmd = findShasum() // this function returns `/usr/bin/shasum` or `/usr/bin/core_perl/shasum` depending on which one exists
command +=
  ' && ' +
  installationFileDescriptors
    .map(({filename, sha256}) => `/usr/bin/echo "${sha256}  ${path.join(tmp, filename)}" | ${shasumCmd} -a 256 -c`)
```